packages/api/src/routes/internal/ehr/canvas/setup.ts

```diff
@@ -1,33 +1,97 @@
+import {
+  canvasDashSource,
+  canvasWebhookSource,
+} from "@metriport/shared/interface/external/ehr/canvas/jwt-token";
+import { EhrSources } from "@metriport/shared/interface/external/ehr/source";
 import { Request, Response } from "express";
 import Router from "express-promise-router";
 import httpStatus from "http-status";
+import { nanoid } from "nanoid";
 import { updateCustomerBillingToPointToParent } from "../../../../command/internal-server/update-customer";
+import { findOrCreateCxMapping } from "../../../../command/mapping/cx";
+import { findOrCreateFacilityMapping } from "../../../../command/mapping/facility";
+import { saveJwtToken } from "../../../../external/ehr/shared/utils/jwt-token";
 import { requestLogger } from "../../../helpers/request-logger";
 import { getUUIDFrom } from "../../../schemas/uuid";
-import { asyncHandler, getFromQueryOrFail } from "../../../util";
+import { asyncHandler, getFromQuery, getFromQueryOrFail } from "../../../util";
 
 const router = Router();
 
+function generateToken(): string {
+  return nanoid(30);
+}
+
+// Calculate expiration (2 years from now in milliseconds)
+function getDefaultExpiration(): number {
+  return Date.now() + 2 * 365 * 24 * 60 * 60 * 1000;
+}
+
 /**
  * POST /internal/ehr/canvas/setup
  *
- * Updates customer billing to point to parent organization
- * @param req.query.parentName - The parent customer's name
+ * Setup a new cx in Canvas
  * @param req.query.childCxId - The child customer's ID
- * @returns 200 OK
+ * @param req.query.facilityId - The facility ID
+ * @param req.query.externalId - The external ID
+ * @param req.query.isTenant - Whether this is a tenant of the main Canvas cx (defaults to true)
+ * @returns 200 OK with the dash and webhook tokens
  */
 router.post(
   "/",
   requestLogger,
   asyncHandler(async (req: Request, res: Response) => {
-    const parentName = getFromQueryOrFail("parentName", req);
     const childCxId = getUUIDFrom("query", req, "childCxId").orFail();
+    const facilityId = getUUIDFrom("query", req, "facilityId").orFail();
+    const externalId = getFromQueryOrFail("externalId", req);
+    const isTenant = getFromQuery("isTenant", req) !== "false"; // defaults to true
+
+    // Only update billing if this is a tenant of the main Canvas cx
+    if (isTenant) {
+      await updateCustomerBillingToPointToParent({ parentName: EhrSources.canvas, childCxId });
+    }
+
+    await findOrCreateFacilityMapping({
+      cxId: childCxId,
+      facilityId: facilityId,
+      externalId: externalId,
+      source: EhrSources.canvas,
+    });
+
+    await findOrCreateCxMapping({
+      cxId: childCxId,
+      source: EhrSources.canvas,
+      externalId: externalId,
+      secondaryMappings: null,
+    });
```

Comment:
Verified this is the correct step 2, to POST `/internal/cx-mapping?cxId=$cxId&source=canvas&externalId=$practiceName`